akka-actor/src/main/scala/akka/util/ImmutableIntMap.scala

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.util
+import java.util.Arrays
+
+import akka.annotation.InternalApi
+
+import scala.annotation.tailrec
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object ImmutableIntMap {
+  def empty: ImmutableIntMap =
+    new ImmutableIntMap(Array.emptyIntArray, Array.empty)
+
+  private val MaxScanLength = 10
+}
+
+/**
+ * INTERNAL API
+ * Specialized Map for primitive `Int` keys to avoid allocations (boxing).
+ * Keys and values are backed by arrays and lookup is performed with binary
+ * search. It's intended for rather small (<1000) maps.
+ */
+@InternalApi private[akka] class ImmutableIntMap private (
+  private val keys: Array[Int], private val values: Array[Int]) {
```

Comment:
@zaharidichev Good point. We could make them private[this] and expose two read-only views (i.e. Iterable?)